src/components/menu/Menu.tsx

```diff
@@ -0,0 +1,114 @@
+    options: any[];
+    buttonLabel: string;
+    buttonColor?: string;
+    buttonHoverColor?: string;
+    componentColor?: string;
+}
+
+function MenuHandler(variant:string,  options:any[], buttonLabel:string, buttonColor:string, 
+  buttonHoverColor:string, componentColor:string) {
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  if (variant === 'radiomenu') {
+		return (
+        <div>
+          <Button 
+            variant="primary" 
+            label={buttonLabel} 
+            size={"large"} 
+            color={buttonColor}
+            hoverColor={buttonHoverColor}
+            onClick={handleClick}></Button>
+
+          <MUIMenu
+            id="basic-menu"
+            anchorEl={anchorEl}
+            open={open}
+            onClose={handleClose}
+            MenuListProps={{
+              'aria-labelledby': 'basic-button',
+            }}>
+            <FormControl>
+                <RadioGroup>
+                    {options.map((option) => (
+                        <MenuItem
+                          key={option}
+                          onClick={handleClose}>
+                              <FormControlLabel 
+                              value={option} 
+                              control={<RadioButton value={option} color={componentColor}/>}
+                              label={option}/>
+                        </MenuItem>
+                    ))}
+              </RadioGroup>
+          </FormControl>
+        </MUIMenu>
+      </div>
+		);}
+
+  else {
+    return (
+      <div>
+        <Button 
+            variant="primary" 
+            label={buttonLabel} 
+            size={"large"} 
+            color={buttonColor}
+            hoverColor={buttonHoverColor}
+            onClick={handleClick}></Button>
+        <MUIMenu
+            id="basic-menu"
+            anchorEl={anchorEl}
+            open={open}
+            onClose={handleClose}
+            MenuListProps={{
+              'aria-labelledby': 'basic-button',
+            }}>
+                {options.map((option) => (
+                    <MenuItem
+                        key={option}
+                        onClick={handleClose}>
+                        {(variant === 'default') ? <div>{option}</div> : null}
+                        {(variant === 'checkmenu') ? <CheckBox label={option} color={componentColor}/> : null}
+                        {(variant === 'togglemenu') ? <div><Toggle color={componentColor}/>{option}</div> : null} 
+                  </MenuItem>
+              ))}
+        </MUIMenu>
+      </div>
+    );
+}}
+
+const Menu = ({
+    variant='default',
+    options=[],
+    buttonLabel="MENU",
+    buttonColor='black',
```

Comment:
I would recommend removing the button color defaults. By default it if color isn't provided it will use the ARGO color scheme.

Reply:
Removed the button altogether.